IRremoteESP8266.cpp

```diff
@@ -362,30 +362,45 @@ void IRsend::sendSAMSUNG(unsigned long data, int nbits)
 void IRsend::mark(int time) {
   // Sends an IR mark for the specified number of microseconds.
   // The mark output is modulated at the PWM frequency.
-  long beginning = micros();
-  while(micros() - beginning < time){
+  int  timeLow = PeriodicTime-timeHigh; 
+  // this logic allows go back to right frequence at integer calc like 40Mhz and allow change duty cicle 
+  long timeEnd = micros()+time; 
+  while(micros() < timeEnd){
     digitalWrite(IRpin, HIGH);
-    delayMicroseconds(halfPeriodicTime);
+    delayMicroseconds(timeHigh );
     digitalWrite(IRpin, LOW);
-    delayMicroseconds(halfPeriodicTime); //38 kHz -> T = 26.31 microsec (periodic time), half of it is 13
-  }
+    delayMicroseconds(timeLow); //38 kHz -> T = 26.31 microsec (periodic time), half of it is 13
+   }
```

Comment:
I verified on LIRCS we have the duty_cicle defined , them I implemented here to test , it is the time on versus off in %, 
Sounds some IR Receptors espects this or maybe this is used to have more power , on tests the value 1/3 33% or 1/2 50% sounds ok